### DISCUSSION TITLE
Why mesh is moving?

### DISCUSSION THREAD

**Discussion by galibubc** [idaholab/moose#23728] [Q&A Modules: Phase field] [answered]
I use the phase field and tensor mechanics module to solve a dendrite formation problem. I have a fundamental question regarding the mesh grids. As we know phase field is an implicit interface-capturing approach; hence we do not need to move the mesh while solving a phase field problem. But when I use the above two modules together **with eigen strain**, it appears that the mesh is moving during the dendrite formation. The evolution of the order parameter alone should be sufficient to track the growth of the dendrites and the interface. I am confused regarding the necessity of moving the mesh simultaneously. Using a fixed grid in the spatial configuration should be sufficient to capture the involved dynamics of the problem. 

I used ` [./TensorMechanics]
    use_displaced_mesh = true
  [../]` to calculate the strains and stresses in the deformed configuration. 

Please comment on the appropriate usage of the above function for such applications. **When I do not use eigen strain, The mesh is not moving.**

![image](https://user-images.githubusercontent.com/121526655/225196523-f985a200-ce7d-4899-8485-717278cc53a0.png)

**Comment by GiudGiud** [comment]
Hello

So you want to track the stresses and strains in the deformed configuration, but without displacing the mesh?

Guillaume

**Comment by galibubc** [comment]
Hello,

Please take a look at the attached image here. The difference with the previous image is that there is no eigen strain in this case.  

I want to measure cauchy stress in a deformed configuration (Eulerian mesh/ fixed grid mesh), not a lagrangian configuration (PK stress). If we calculate cauchy stress with phase-field, we evolve (another equation in space and time) strain tensor (for example: left cauchy green tensor B = FF^T. Using phase field, we avoid the computationally expensive task of moving mesh. 


I am adding my input file here. Please guide me to get the appropriate setting.
[undeformed_mesh.txt](https://github.com/idaholab/moose/files/11004921/undeformed_mesh.txt)
[deformed_mesh.txt](https://github.com/idaholab/moose/files/11004972/deformed_mesh.txt)

![image](https://user-images.githubusercontent.com/121526655/225462008-85136370-728a-4856-ae00-594d75ce5349.png)

**Comment by GiudGiud** [comment]
@dschwen @laagesen on phase field

**Reply by laagesen** [reply, marked as answer]
When you solve the mechanical equilibrium equation, you solve for local displacement at each position (and local strains are derivatives of the displacement fields) When you visualize the mesh, depending on the tool you use, the software may display the elements as moved according to the local displacement field. This is usually a user-adjustable parameter, so you can switch it off or you can displace the mesh by some scalar multiple of the displacement field. When you don't include solving the mechanical equilibrium simultaneously, there are no displacement fields.

**Reply by galibubc** [reply]
1. Does that mean `[./TensorMechanics] use_displaced_mesh = false[../]` will give the same results for the primitive variables (disp_x, disp_y, eta) without a deformed mesh in the visualization tool.  

2. MOOSE uses the tensor mechanics module to solve for the mechanical equilibrium. Is the module solving for displacement or velocity as the primitive variable?

3. For clarification, when we do not move the mesh, does the solver still compute the cauchy stresses in a deformed configuration? For this to happen, do you evolve any of the deformation tensors with space and time in the eulerian configuration?

4. Also, why it was not deforming the mesh when I used `[./TensorMechanics] use_displaced_mesh = true[../],` in the previous figure. The only difference between the two figures is that one has eigen strain (-0.32, **deformed mesh**) other does not (0.0, **undeformed mesh**).

**Reply by GiudGiud** [reply]
For 2) displacement variables are used as nonlinear variables, not velocities

For 1 & 3 i would encourage you to check the documentation for the tensor mechanics module more closely.
@maxnezdyur will know for sure otherwise

**Reply by galibubc** [reply]
Thanks for the response. I checked the documents, but it is not clear to me. My apologies

I am revising the questions again.
 
1. Why it was not deforming the mesh when I used `[./TensorMechanics] use_displaced_mesh = true[../]` in the previous figure. The only difference between the two figures is that one has eigen strain (-0.32, **deformed mesh**) other does not (0.0, **undeformed mesh**).

2. Does that mean `[./TensorMechanics] use_displaced_mesh = false[../]` will give the same results for the primitive variables (disp_x, disp_y, eta) without a deformed mesh in the visualization tool

3. For clarification, when we do not move the mesh, does the solver still compute the cauchy stresses in a deformed configuration?

**Reply by GiudGiud** [reply]
@recuero
